### ISSUE TITLE
challenge update 1

### ISSUE THREAD

**Issue by mikedeverna** [MacroFit/iOS#57]
Please integrate ability for a video in the challenge description.  In the v2 design, only implement for now the ability to start and challenge in the case that it's a timer (countdown) type challenge and integrate this challenge first.  

Please also see if there's an api for machine voices we can use to countdown at 3,2,1 or various intervals.

**Comment by munishbansal** [comment]
@mikedeverna I am listing down the changes needed. Let me know if there is any gap:

1. Submit Score Button shall be replaced by Start Workout

2. Challenge video to be added in the challenge description screen

3. On Tapping "Start Workout" button, this screen should open:  
https://projects.invisionapp.com/share/9GP0WNWW7D2#/screens/334778517

4. User should be allowed to Set a pacer goal

5. User will not enter score. The score is the time duration between "Start workout" and "end workout" events. The lesser the time the better is the score. 

Please confirm. 

By Tuesday the new Home screen shall be completed. Will start on the challenge post that.

**Comment by mikedeverna** [comment]
For #3) it's similar to that screen although since it's a timer down it would start at the time limit of the challenge 

4) I don't think pacer goal is spec'd out enough.  Is that a much bigger effort?  Maybe we should chat specifically about what it does or leave that for another ticket.  For example.

Challenge:  1 min pushup exercise
Description: perform as many pushups in 1 min as possible
Pacer Scenario:  User inputs goal of 45 reps.  What does the pacer do?  Audio pacing would be most helpful.  At what time intervals would the audio voice say "30 seconds, target pace should be 22 reps"

5) This does not make sense.  Since this is a countdown (timer) format, user will still need to enter reps

6) Audio.  Lets please discuss first a scalable audio voice solution or direction.
a) Automated - we could use some api of voice commands, this would need to be an acceptable sounding voice
b) voiceover, we would have to provide the voice
c) we would need to set some default audio settings per workout / challenge format

**Comment by munishbansal** [comment]
@mikedeverna For (3) 

You said there are 5 type of challenges:
1. TimeBased
2. AMRAP
3. TimerUp
4. Tabata
5. SetTimeBased

the one we are talking about here is: AMRAP?

Also, at the moment the fields in a challenge are:
              "title": "required|string|max:50",
              "format": "nullable|in:TimeBased,AMRAP,TimerUp,Tabata,SetTimeBased",
              "description": "required|string|max:250",
              "photo": "nullable|string|max:250",
              "video": "nullable|string|max:250",
              "tags": "required|string|max:250",
              "score_unit": "nullable|string|max:250",
              "score_question": "nullable|string|max:250",
              "exercises": "nullable|string|max:1250",
              "participants_count": "required",
              "is_scoring_in_time": "required|boolean",
              "the_more_the_better": "required|boolean"

I believe we need to add a field: time_limit  in the type of challenges in which the timer will start counting down from pre-set number. Please suggest. 

4. We can pacer goal as a separate ticket if its not spec'd out enough. Its purely frontend work and that is not a trivial effort. 

5. Ok. Understood now. Good that I asked. Because my understanding was entirely different :)

6. We can take it up along with 4. Pls allow us to do some research on this in parallel. 

Thanks.

**Comment by munishbansal** [comment]
@mikedeverna 
### a) Automated should be possible. PoC needed

```
Audio.
a) Automated - we could use some api of voice commands, this would need to be an acceptable sounding voice
b) voiceover, we would have to provide the voice
c) we would need to set some default audio settings per workout / challenge format

```
